lib/std/Io.zig

```diff
@@ -548,8 +548,1081 @@ pub fn PollFiles(comptime StreamEnum: type) type {
+    /// `await` will be a no-op.
+    ///
+    /// Thread-safe.
+    async: *const fn (
+        /// Corresponds to `Io.userdata`.
+        userdata: ?*anyopaque,
+        /// The pointer of this slice is an "eager" result value.
+        /// The length is the size in bytes of the result type.
+        /// This pointer's lifetime expires directly after the call to this function.
+        result: []u8,
+        result_alignment: std.mem.Alignment,
+        /// Copied and then passed to `start`.
+        context: []const u8,
+        context_alignment: std.mem.Alignment,
+        start: *const fn (context: *const anyopaque, result: *anyopaque) void,
+    ) ?*AnyFuture,
+    /// Thread-safe.
+    concurrent: *const fn (
+        /// Corresponds to `Io.userdata`.
+        userdata: ?*anyopaque,
+        result_len: usize,
+        result_alignment: std.mem.Alignment,
+        /// Copied and then passed to `start`.
+        context: []const u8,
+        context_alignment: std.mem.Alignment,
+        start: *const fn (context: *const anyopaque, result: *anyopaque) void,
+    ) error{OutOfMemory}!*AnyFuture,
+    /// This function is only called when `async` returns a non-null value.
+    ///
+    /// Thread-safe.
+    await: *const fn (
+        /// Corresponds to `Io.userdata`.
+        userdata: ?*anyopaque,
+        /// The same value that was returned from `async`.
+        any_future: *AnyFuture,
+        /// Points to a buffer where the result is written.
+        /// The length is equal to size in bytes of result type.
+        result: []u8,
+        result_alignment: std.mem.Alignment,
+    ) void,
+    /// Equivalent to `await` but initiates cancel request.
+    ///
+    /// This function is only called when `async` returns a non-null value.
+    ///
+    /// Thread-safe.
+    cancel: *const fn (
+        /// Corresponds to `Io.userdata`.
+        userdata: ?*anyopaque,
+        /// The same value that was returned from `async`.
+        any_future: *AnyFuture,
+        /// Points to a buffer where the result is written.
+        /// The length is equal to size in bytes of result type.
+        result: []u8,
+        result_alignment: std.mem.Alignment,
+    ) void,
+    /// Returns whether the current thread of execution is known to have
+    /// been requested to cancel.
+    ///
+    /// Thread-safe.
+    cancelRequested: *const fn (?*anyopaque) bool,
+
+    /// Executes `start` asynchronously in a manner such that it cleans itself
+    /// up. This mode does not support results, await, or cancel.
+    ///
+    /// Thread-safe.
+    groupAsync: *const fn (
+        /// Corresponds to `Io.userdata`.
+        userdata: ?*anyopaque,
+        /// Owner of the spawned async task.
+        group: *Group,
+        /// Copied and then passed to `start`.
+        context: []const u8,
+        context_alignment: std.mem.Alignment,
+        start: *const fn (*Group, context: *const anyopaque) void,
```

Comment:
Is there a particular reason for passing the `*Group` as a separate parameter to `start` (which is not needed by the majority of call sites) instead of just having `Select(...).async` add it to the args in `context`?

Reply:
maybe I'm not understanding the suggestion, but that kind of sounds like a pain in the ass and unclear that it will generate better code